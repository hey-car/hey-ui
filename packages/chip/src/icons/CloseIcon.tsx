import React from 'react';

import styles from '../styles/icon.module.css';

interface CloseIconProps {
  /**
   * 	boolean to make chip disable.
   */
  disabled?: boolean;
}

function CloseIcon({ disabled = false }: CloseIconProps) {
  return (
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className={disabled ? styles.disabled : styles.active}
        clip-rule="evenodd"
        d="M8.9428 7.99999L13.8047 3.13806L12.8619 2.19525L7.99999 7.05718L3.13806 2.19525L2.19525 3.13806L7.05718 7.99999L2.19525 12.8619L3.13806 13.8047L7.99999 8.9428L12.8619 13.8047L13.8047 12.8619L8.9428 7.99999Z"
        fill-rule="evenodd"
      />
    </svg>
  );
}

export default CloseIcon;
