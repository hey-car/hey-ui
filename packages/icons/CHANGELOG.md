# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.3.0](https://github.com/hey-car/heycar-uikit/compare/@heycar-uikit/icons@1.2.0...@heycar-uikit/icons@1.3.0) (2022-08-03)


### Features

* **badge:** The first implantation of badge component ([#42](https://github.com/hey-car/heycar-uikit/issues/42)) ([2ca5a7e](https://github.com/hey-car/heycar-uikit/commit/2ca5a7ea897c4cdeccaadca4f4d1592be40fe6e7))





# [1.2.0](https://github.com/hey-car/heycar-uikit/compare/@heycar-uikit/icons@1.1.0...@heycar-uikit/icons@1.2.0) (2022-08-02)


### Features

* **icons:** the first implementation of the icons package ([#39](https://github.com/hey-car/heycar-uikit/issues/39)) ([74fd00d](https://github.com/hey-car/heycar-uikit/commit/74fd00d9227e3e915fbb54962d56ab9d64715b5c))
* **icons:** updated icons ([#45](https://github.com/hey-car/heycar-uikit/issues/45)) ([2293b7a](https://github.com/hey-car/heycar-uikit/commit/2293b7ad0135fb645495afb9a11583ce8ef7e6e3))
* **switch:** initial implementation of the switch component ([#41](https://github.com/hey-car/heycar-uikit/issues/41)) ([fae4001](https://github.com/hey-car/heycar-uikit/commit/fae40014e387fd6a84275fdee825b265b798b7d3))





# [1.1.0](https://github.com/hey-car/heycar-uikit/compare/@heycar-uikit/icons@1.1.0...@heycar-uikit/icons@1.1.0) (2022-07-27)


### Features

* **icons:** the first implementation of the icons package ([#39](https://github.com/hey-car/heycar-uikit/issues/39)) ([74fd00d](https://github.com/hey-car/heycar-uikit/commit/74fd00d9227e3e915fbb54962d56ab9d64715b5c))
* **switch:** initial implementation of the switch component ([#41](https://github.com/hey-car/heycar-uikit/issues/41)) ([fae4001](https://github.com/hey-car/heycar-uikit/commit/fae40014e387fd6a84275fdee825b265b798b7d3))





# [1.1.0](https://github.com/hey-car/heycar-uikit/compare/@heycar-uikit/icons@1.1.0...@heycar-uikit/icons@1.1.0) (2022-07-25)


### Features

* **icons:** the first implementation of the icons package ([#39](https://github.com/hey-car/heycar-uikit/issues/39)) ([74fd00d](https://github.com/hey-car/heycar-uikit/commit/74fd00d9227e3e915fbb54962d56ab9d64715b5c))





# 1.1.0 (2022-07-22)


### Features

* **icons:** the first implementation of the icons package ([#40](https://github.com/hey-car/heycar-uikit/issues/40)) ([60e2055](https://github.com/hey-car/heycar-uikit/commit/60e205578ce2d2024eb22c0a8384a17d7fbe4b91))
