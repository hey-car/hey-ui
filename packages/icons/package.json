{
  "name": "@heycar-uikit/icons",
  "version": "1.7.0",
  "description": "HeyCar-UIKit icons",
  "main": "dist/index.js",
  "module": "./dist/esm/index.js",
  "keywords": [
    "ui",
    "uikit",
    "react",
    "components",
    "heycar",
    "typescript",
    "ui-components",
    "design-systems",
    "icons"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hey-car/heycar-uikit.git"
  },
  "homepage": "https://github.com/hey-car/heycar-uikit/tree/main/packages/icons#readme",
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^17.0.1"
  },
  "dependencies": {
    "classnames": "^2.3.1"
  },
  "postinstall": "node ./dist/send-stats.js > /dev/null 2>&1 || exit 0",
  "author": "HeyCar Team",
  "license": "UNLICENSED",
  "gitHead": "58e9efe350cbc1d37ba37590f55347f2fd23ad13"
}
