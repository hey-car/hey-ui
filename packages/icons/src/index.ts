export * from './paths/ABSport';
export * from './paths/AC';
export * from './paths/AndroidAuto';
export * from './paths/ApplePlay';
export * from './paths/Automatic';
export * from './paths/Buying–°ar';
export * from './paths/Calculator';
export * from './paths/Calendar';
export * from './paths/Call';
export * from './paths/Car';
export * from './paths/Card';
export * from './paths/Clock';
export * from './paths/Close';
export * from './paths/Copy';
export * from './paths/Collect';
export * from './paths/Convertible';
export * from './paths/CruiseControl';
export * from './paths/Coupe';
export * from './paths/Dealer';
export * from './paths/ElectricWindow';
export * from './paths/Email';
export * from './paths/Engine';
export * from './paths/Estate';
export * from './paths/Exterior';
export * from './paths/EV';
export * from './paths/Eye';
export * from './paths/Filter';
export * from './paths/Finance';
export * from './paths/FiveDoors';
export * from './paths/Fuel';
export * from './paths/Gallery';
export * from './paths/Hamburger';
export * from './paths/Hatchback';
export * from './paths/HeatedMirror';
export * from './paths/HeatedSeat';
export * from './paths/Horsepower';
export * from './paths/Hybrid';
export * from './paths/Image';
export * from './paths/Info';
export * from './paths/Interior';
export * from './paths/ISOFix';
export * from './paths/Key';
export * from './paths/LargeBoot';
export * from './paths/Leather';
export * from './paths/List';
export * from './paths/Location';
export * from './paths/Lorry';
export * from './paths/LowEmission';
export * from './paths/LumbarSupport';
export * from './paths/Manual';
export * from './paths/MBG';
export * from './paths/Mileage';
export * from './paths/Minus';
export * from './paths/MinusCircle';
export * from './paths/MVP';
export * from './paths/NewCar';
export * from './paths/Others';
export * from './paths/ParkingCamera';
export * from './paths/Play';
export * from './paths/PlusCircle';
export * from './paths/Pickup';
export * from './paths/Read';
export * from './paths/RegNr';
export * from './paths/RegularBoot';
export * from './paths/ReverseSensor';
export * from './paths/Review';
export * from './paths/Saloon';
export * from './paths/Satnav';
export * from './paths/Search';
export * from './paths/Seats';
export * from './paths/Share';
export * from './paths/ShoppingBag';
export * from './paths/Sort';
export * from './paths/Sound';
export * from './paths/SoundSystem';
export * from './paths/Sunroof';
export * from './paths/SUV';
export * from './paths/Swap';
export * from './paths/ThreeDoors';
export * from './paths/ThreeSixty';
export * from './paths/Tick';
export * from './paths/TickCircle';
export * from './paths/Trend';
export * from './paths/USB';
export * from './paths/Video';
export * from './paths/VideoReq';
export * from './paths/Van';
export * from './paths/Warning';
export * from './paths/Warranty';
export * from './paths/XLBoot';
export * from './paths/Facebook';
export * from './paths/Instagram';
export * from './paths/Twitter';
export * from './paths/LinkedIn';
export * from './paths/Pinterest';
export * from './paths/TikTok';
export * from './paths/Messenger';
export * from './paths/Youtube';
export * from './paths/Whatsapp';
export * from './paths/ArrowDown';
export * from './paths/ArrowTop';
export * from './paths/ArrowLeft';
export * from './paths/ArrowRight';
export * from './paths/ChevronDown';
export * from './paths/ChevronTop';
export * from './paths/ChevronLeft';
export * from './paths/ChevronRight';
export * from './paths/Lock';
export * from './paths/LockOpen';
export * from './paths/OfferLeft';
export * from './paths/OfferRight';
