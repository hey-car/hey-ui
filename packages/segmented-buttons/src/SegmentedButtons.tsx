/* eslint-disable react-hooks/exhaustive-deps */
import React from 'react';

import { SegmentedButtonsProps } from './SegmentedButtons.types';

import styles from './styles/default.module.css';

const SegmentedButtons = React.forwardRef<
  HTMLDivElement,
  SegmentedButtonsProps
>(
  (
    {
      className,
      currentValue,
      buttons,
      dataTestId,
      onChange,
      wideView,
      ...restProps
    },
    ref,
  ) => {
    return (
      <div
        className={`${styles.wrapper} ${
          wideView ? styles.isWideView : ''
        } ${className}`}
        data-test-id={dataTestId}
        ref={ref}
        role="radiogroup"
        {...restProps}
      >
        {buttons.map(btn => {
          const isActive = currentValue === btn.value;

          return (
            <button
              aria-checked={isActive}
              aria-label={btn.label}
              data-test-id={btn.dataTestId}
              disabled={btn.isDisabled}
              key={btn.label}
              onClick={() => {
                if (!isActive) onChange(btn.value);
              }}
              role="radio"
            >
              {btn.icon}
              {btn.label}
            </button>
          );
        })}
      </div>
    );
  },
);

SegmentedButtons.displayName = 'SegmentedButtons';

export default SegmentedButtons;
