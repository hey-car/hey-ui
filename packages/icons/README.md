# HeyCar-UIKit - Icons

[![Tests](https://github.com/hey-car/heycar-uikit/actions/workflows/build.yml/badge.svg)](https://github.com/hey-car/heycar-uikit/actions/workflows/build.yml)
[![Coverage Status](https://coveralls.io/repos/github/hey-car/heycar-uikit/badge.svg)](https://coveralls.io/github/hey-car/heycar-uikit)
[![Demo build](https://github.com/hey-car/heycar-uikit/actions/workflows/main.yml/badge.svg)](https://github.com/hey-car/heycar-uikit/actions/workflows/main.yml)
[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)

## Installation

To install and save in your package.json dependencies, run the command below using npm:

```bash
npm install @heycar-uikit/icons
```

## Usage

Semantic vector graphics. Before using icons, you need to install import icons using one of these two options:

- Option 1:

```tsx
import CarIcon from '@heycar-uikit/icons/Car';
import CloseIcon from '@heycar-uikit/icons/Close';

function App() {
  return (
    <div>
      <Button
        leftIcon={<CarIcon />}
        size="large"
        color="primary"
        variant="outlined"
      >
        Hello World
      </Button>
      <CloseIcon />
    </div>
  );
}
```

- Option 2:

```tsx
import { Car, Close } from '@heycar-uikit/icons';

function App() {
  return (
    <div>
      <Button
        leftIcon={<CarIcon />}
        size="large"
        color="primary"
        variant="outlined"
      >
        Hello World
      </Button>
      <CloseIcon />
    </div>
  );
}
```

## Documentation and sandbox

[Storybook](https://hey-car.github.io/heycar-uikit/main/?path=/docs/components-icons--icons) documentation and sandbox.
