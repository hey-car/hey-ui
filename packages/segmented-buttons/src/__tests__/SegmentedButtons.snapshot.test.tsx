/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react';
import { render } from '@testing-library/react';

import { Gallery } from '@heycar-uikit/icons';

import SegmentedButtons from '../SegmentedButtons';

import { BTN_DATA, modifyBtns, TWO_BTNS } from './SegmentedButtons.mocks';

describe('Button Snapshots tests', () => {
  it('should render 2 buttons', () => {
    expect(
      render(
        <SegmentedButtons
          buttons={TWO_BTNS}
          id="sb-test-0"
          onChange={() => {}}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should render 3 buttons', () => {
    expect(
      render(
        <SegmentedButtons
          buttons={BTN_DATA}
          id="sb-test-1"
          onChange={() => {}}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should render buttons with icons', () => {
    const btnsWithIcons = modifyBtns(false, <Gallery />);

    expect(
      render(
        <SegmentedButtons
          buttons={btnsWithIcons}
          id="sb-test-2"
          onChange={() => {}}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should render buttons as disabled', () => {
    const btnsDisabled = modifyBtns(true, undefined);

    expect(
      render(
        <SegmentedButtons
          buttons={btnsDisabled}
          id="sb-test-3"
          onChange={() => {}}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should render an active button', () => {
    expect(
      <SegmentedButtons
        buttons={BTN_DATA}
        currentValue="lft"
        id="sb-test-4"
        onChange={() => {}}
      />,
    ).toMatchSnapshot();
  });

  it('should render an active button disabled', () => {
    const btnsDisabled = modifyBtns(true, undefined);

    expect(
      render(
        <SegmentedButtons
          buttons={btnsDisabled}
          currentValue="lft"
          id="sb-test-5"
          onChange={() => {}}
        />,
      ),
    ).toMatchSnapshot();
  });
});
