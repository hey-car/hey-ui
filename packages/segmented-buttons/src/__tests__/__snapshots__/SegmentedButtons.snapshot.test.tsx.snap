// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots tests should render 2 buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper  undefined"
        id="sb-test-0"
        role="radiogroup"
      >
        <button
          aria-checked="false"
          aria-label="Left"
          class="btn"
          data-test-id="btn-1"
          role="radio"
        >
          Left
        </button>
        <button
          aria-checked="false"
          aria-label="Right"
          class="btn"
          data-test-id="btn-2"
          role="radio"
        >
          Right
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper  undefined"
      id="sb-test-0"
      role="radiogroup"
    >
      <button
        aria-checked="false"
        aria-label="Left"
        class="btn"
        data-test-id="btn-1"
        role="radio"
      >
        Left
      </button>
      <button
        aria-checked="false"
        aria-label="Right"
        class="btn"
        data-test-id="btn-2"
        role="radio"
      >
        Right
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Snapshots tests should render 3 buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper  undefined"
        id="sb-test-1"
        role="radiogroup"
      >
        <button
          aria-checked="false"
          aria-label="Left"
          class="btn"
          role="radio"
        >
          Left
        </button>
        <button
          aria-checked="false"
          aria-label="Middle"
          class="btn"
          role="radio"
        >
          Middle
        </button>
        <button
          aria-checked="false"
          aria-label="Right"
          class="btn"
          role="radio"
        >
          Right
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper  undefined"
      id="sb-test-1"
      role="radiogroup"
    >
      <button
        aria-checked="false"
        aria-label="Left"
        class="btn"
        role="radio"
      >
        Left
      </button>
      <button
        aria-checked="false"
        aria-label="Middle"
        class="btn"
        role="radio"
      >
        Middle
      </button>
      <button
        aria-checked="false"
        aria-label="Right"
        class="btn"
        role="radio"
      >
        Right
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Snapshots tests should render an active button 1`] = `
<SegmentedButtons
  buttons={
    Array [
      Object {
        "icon": undefined,
        "isDisabled": true,
        "label": "Left",
        "value": "lft",
      },
      Object {
        "icon": undefined,
        "isDisabled": true,
        "label": "Middle",
        "value": "mid",
      },
      Object {
        "icon": undefined,
        "isDisabled": true,
        "label": "Right",
        "value": "rht",
      },
    ]
  }
  currentValue="lft"
  id="sb-test-4"
  onChange={[Function]}
/>
`;

exports[`Button Snapshots tests should render an active button disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper  undefined"
        id="sb-test-5"
        role="radiogroup"
      >
        <button
          aria-checked="true"
          aria-label="Left"
          class="btn"
          disabled=""
          role="radio"
        >
          Left
        </button>
        <button
          aria-checked="false"
          aria-label="Middle"
          class="btn"
          disabled=""
          role="radio"
        >
          Middle
        </button>
        <button
          aria-checked="false"
          aria-label="Right"
          class="btn"
          disabled=""
          role="radio"
        >
          Right
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper  undefined"
      id="sb-test-5"
      role="radiogroup"
    >
      <button
        aria-checked="true"
        aria-label="Left"
        class="btn"
        disabled=""
        role="radio"
      >
        Left
      </button>
      <button
        aria-checked="false"
        aria-label="Middle"
        class="btn"
        disabled=""
        role="radio"
      >
        Middle
      </button>
      <button
        aria-checked="false"
        aria-label="Right"
        class="btn"
        disabled=""
        role="radio"
      >
        Right
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Snapshots tests should render buttons as disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper  undefined"
        id="sb-test-3"
        role="radiogroup"
      >
        <button
          aria-checked="false"
          aria-label="Left"
          class="btn"
          disabled=""
          role="radio"
        >
          Left
        </button>
        <button
          aria-checked="false"
          aria-label="Middle"
          class="btn"
          disabled=""
          role="radio"
        >
          Middle
        </button>
        <button
          aria-checked="false"
          aria-label="Right"
          class="btn"
          disabled=""
          role="radio"
        >
          Right
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper  undefined"
      id="sb-test-3"
      role="radiogroup"
    >
      <button
        aria-checked="false"
        aria-label="Left"
        class="btn"
        disabled=""
        role="radio"
      >
        Left
      </button>
      <button
        aria-checked="false"
        aria-label="Middle"
        class="btn"
        disabled=""
        role="radio"
      >
        Middle
      </button>
      <button
        aria-checked="false"
        aria-label="Right"
        class="btn"
        disabled=""
        role="radio"
      >
        Right
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Snapshots tests should render buttons with icons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper  undefined"
        id="sb-test-2"
        role="radiogroup"
      >
        <button
          aria-checked="false"
          aria-label="Left"
          class="btn"
          role="radio"
        >
          <svg
            class="svgIcon inherit"
            data-test-id="GalleryIcon"
            focusable="false"
            style="font-size: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
              fill-rule="evenodd"
            />
          </svg>
          Left
        </button>
        <button
          aria-checked="false"
          aria-label="Middle"
          class="btn"
          role="radio"
        >
          <svg
            class="svgIcon inherit"
            data-test-id="GalleryIcon"
            focusable="false"
            style="font-size: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
              fill-rule="evenodd"
            />
          </svg>
          Middle
        </button>
        <button
          aria-checked="false"
          aria-label="Right"
          class="btn"
          role="radio"
        >
          <svg
            class="svgIcon inherit"
            data-test-id="GalleryIcon"
            focusable="false"
            style="font-size: 24px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
              fill-rule="evenodd"
            />
          </svg>
          Right
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper  undefined"
      id="sb-test-2"
      role="radiogroup"
    >
      <button
        aria-checked="false"
        aria-label="Left"
        class="btn"
        role="radio"
      >
        <svg
          class="svgIcon inherit"
          data-test-id="GalleryIcon"
          focusable="false"
          style="font-size: 24px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
            fill-rule="evenodd"
          />
        </svg>
        Left
      </button>
      <button
        aria-checked="false"
        aria-label="Middle"
        class="btn"
        role="radio"
      >
        <svg
          class="svgIcon inherit"
          data-test-id="GalleryIcon"
          focusable="false"
          style="font-size: 24px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
            fill-rule="evenodd"
          />
        </svg>
        Middle
      </button>
      <button
        aria-checked="false"
        aria-label="Right"
        class="btn"
        role="radio"
      >
        <svg
          class="svgIcon inherit"
          data-test-id="GalleryIcon"
          focusable="false"
          style="font-size: 24px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 19L19 19L19 5L5 5L5 19ZM19 21L21 21V19L21 5L21 3L19 3L5 3H3L3 5L3 19L3 21L5 21L19 21ZM17 17H13L13 13L17 13L17 17ZM13 11L17 11V7L13 7L13 11ZM11 17L7 17L7 13L11 13L11 17ZM7 11L11 11L11 7L7 7L7 11Z"
            fill-rule="evenodd"
          />
        </svg>
        Right
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
